### PR TITLE
Registry sync v2 - API

### PR DESCRIPTION
**Pull request by thecatcore**
New registry api, more extensible, only handles registry infrastructure (remapping included).

Individual registries for content are moved to their own module in other pr depending on this one.

### CODE REVIEW

File: legacy-fabric-registry-sync-api-v2/common/src/main/java/net/legacyfabric/fabric/api/registry/v2/registry/holder/Registry.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import net.legacyfabric.fabric.api.event.Event;
+import net.legacyfabric.fabric.api.registry.v2.event.RegistryBeforeAddCallback;
+import net.legacyfabric.fabric.api.registry.v2.event.RegistryEntryAddedCallback;
+import net.legacyfabric.fabric.api.util.Identifier;
+
+public interface Registry<T> extends Iterable<T> {
```
Comment (BoogieMonster1O1) [inline legacy-fabric-registry-sync-api-v2/common/src/main/java/net/legacyfabric/fabric/api/registry/v2/registry/holder/Registry.java:28]: Registry -> RegistryExtensions? I don't think its a good idea to keep it as the same name as an existing minecraft class

**Review by BoogieMonster1O1 — CHANGES_REQUESTED** [review]

File: legacy-fabric-api-base/common/src/main/java/net/legacyfabric/fabric/api/util/Identifier.java [hunk 2]
```diff
@@ -71,6 +71,10 @@ public String toString() {
 		return this.namespace + ':' + this.path;
 	}
 
+	public String toTranslationKey() {
+		return this.namespace + "." + this.path;
```
[the remark this answers is not in the document]
Comment (thecatcore) [reply legacy-fabric-api-base/common/src/main/java/net/legacyfabric/fabric/api/util/Identifier.java:75]: It's used later in the registry implementation to set the translation key automatically, which was something already done in registry sync v1 previously.

File: legacy-fabric-registry-sync-api-v2/common/src/main/java/net/legacyfabric/fabric/api/registry/v2/registry/holder/Registry.java (hunk 1 above)
Comment (thecatcore) [reply legacy-fabric-registry-sync-api-v2/common/src/main/java/net/legacyfabric/fabric/api/registry/v2/registry/holder/Registry.java:28]: `RegistryExtensions` doesn't fit in this case as this interface can be used to implement a custom registry implementation.
However maybe we should prefix those classes with either `Fabric`, `LF` or `LegacyFabric` to differentiate them from vanilla classes as you've said.

Comment (thecatcore) [reply legacy-fabric-registry-sync-api-v2/common/src/main/java/net/legacyfabric/fabric/api/registry/v2/registry/holder/Registry.java]: Renamed to `FabricRegistry`.

**Review by ENDERZOMBI102 — APPROVED** [review]
Looks good!
Naming seems correct too

**Pull request merged**